File: 20A40/cir-Israel/Ejercicio 40/Employee.cpp
#include "Employee.h"

// Constructor
Employee::Employee(std::string name, std::string address, float salary) {
    this->name = name;
    this->address = address;
    this->salary = salary;
}

void Employee::setName(std::string name) {
    this->name = name;
}

void Employee::setAddress(std::string address) {
    this->address = address;
}

void Employee::setSalary(float salary) {
    this->salary = salary;
}

std::string Employee::getName() const { return name;}
std::string Employee::getAddress() const { return address; }
float Employee::getSalary() const { return salary; }

// Método para recibir salario
void Employee::receivePay() {
}

std::string Employee::toString() const {
    return "Employee{name='" + name + "', " +
           "address='" + address + "', " +
           "salary=" + std::to_string(salary) + "}";
}

File: 20A40/cir-Israel/Ejercicio 36/Employee.java
public class Employee {              // mutable 
  private String name;   
  private double salary;
  private String birthDate;
  //private String place;           // puesto laboral

  public Employee() {}             // constructor por defecto
  
  public Employee(String name, double salary, String bd) {
    this.name = name;
    this.salary = salary;
    this.birthDate = bd;
  }

  public String getDetails() {
    return "Name: " + name + "\nSalary: " + salary;
  }
  
  public String getName() {  return name;  }
  public double getSalary() {  return salary;  }
  public String getBirthDate() {  return birthDate;  } 
  //public String getPlace() {  return place;  }
  
  public boolean setName(String n){
    if (n!=null && !n.equals("")) {
      name = n;
      return true;
    } else return false;
  } 
  
  public boolean setBirthDate(String fecha) {
    
    if (fecha.length() > 0) {  // validacion ingenua
       this.birthDate = fecha;
      return true;
    } else return false;
  }
  
  public boolean setSalary(double s) {
    if (s > 0) {
      salary = s;
      return true;
    }
    return false;
  }

  public String toString(){
    String cadString = "Nombre: "+name+"\n";
    cadString += "Salario: "+salary+"\n";
    cadString += "Fecha: "+birthDate+"\n";
    return cadString;

  } 
}
